azext_iot/tests/iothub/devices/test_hub_export_import.py

```diff
@@ -0,0 +1,328 @@
+
+                        self.cmd("iot hub module-identity create -d {} -m deviceModule -n {} -g {} --auth-type={} --am x509_thumbprint --ptp {} --stp {}" \
+                            .format(device, self.entity_name, self.entity_rg, auth_phase, ptp, stp))
+                        
+                    # add a property and a tag to each module's twin
+
+                    val = generate_generic_id()
+                    self.cmd(
+                        f"iot hub module-twin update -d {device} -m deviceModule -n {self.entity_name} -g {self.entity_rg} --auth-type={auth_phase} --set properties.desired.testProp={val}"
+                    )
+
+                    patch_tags = {
+                        generate_generic_id(): generate_generic_id(),
+                        generate_generic_id(): generate_generic_id(),
+                    }
+                    self.kwargs["patch_tags"] = json.dumps(patch_tags)
+
+                    self.cmd(
+                        f"iot hub module-twin update -d {device} -m deviceModule -n {self.entity_name} -g {self.entity_rg} --auth-type={auth_phase}"
+                        " --tags '{patch_tags}'"
+                    )
+
+    def compare_hubs(self):
+
+        # compare configurations (there's only one)
+
+        orig_hub_configs = self.cmd(
+            f"iot hub configuration list -n {self.entity_name} -g {self.entity_rg}"
+        ).get_output_in_json()
+
+        dest_hub_configs = self.cmd(
+            f"iot hub configuration list -n {self.dest_hub} -g {self.dest_hub_rg}"
+        ).get_output_in_json()
+
+        assert(len(orig_hub_configs) == 1 and len(dest_hub_configs) == 1)
+
+        assert(orig_hub_configs[0]["id"] == dest_hub_configs[0]["id"])
+        assert(orig_hub_configs[0]["content"] == dest_hub_configs[0]["content"])
+        assert(orig_hub_configs[0]["metrics"] == dest_hub_configs[0]["metrics"])
+        assert(orig_hub_configs[0]["priority"] == dest_hub_configs[0]["priority"])
+        assert(orig_hub_configs[0]["systemMetrics"]["queries"] == dest_hub_configs[0]["systemMetrics"]["queries"])
+        assert(orig_hub_configs[0]["targetCondition"] == dest_hub_configs[0]["targetCondition"])
+
+        # compare devices
+
+        orig_hub_identities = self.cmd(
+            f"iot hub device-identity list -n {self.entity_name} -g {self.entity_rg}"
+        ).get_output_in_json()
+        dest_hub_identities = self.cmd(
+            f"iot hub device-identity list -n {self.dest_hub} -g {self.dest_hub_rg}"
+        ).get_output_in_json()
+
+        for device in orig_hub_identities:
+            target = None
+            for id in dest_hub_identities:
+                if(device["deviceId"] == id["deviceId"]):
+                    target = id
+                    break
+
+            assert target
+
+            assert (device["authenticationType"] == target["authenticationType"])
```

Comment:
would be helpful to change some of these checks to helper functions. See if device tests already have helper functions and if not, then see if you can use the helper functions you made here for those too!